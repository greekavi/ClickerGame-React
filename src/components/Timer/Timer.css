.Timer{
    font-family: Century Gothic,CenturyGothic,AppleGothic,sans-serif; 
    font-size: 25px;
}