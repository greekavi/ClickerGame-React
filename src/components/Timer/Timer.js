import React,{useState,useEffect} from "react";
import './Timer.css'
function Timer({TimerSeconds}){  
    const [timeOver,setTimeOver]=useState(false);
    const [time,setTime]=useState(true);
    useEffect(()=>{
      
        if(!TimerSeconds)
       {
        setTimeOver(true);
        setTime(false)
       }
        else
        {
        setTime(true);
        setTimeOver(false)
        }
    },[TimerSeconds])
    return(
         <div>
                
        {time&&<b className="Timer">Timer : {TimerSeconds}</b>}
        {timeOver&&  <b className="Timer stop">Time Up!!!</b>}
      </div>
    );
}
export default Timer;