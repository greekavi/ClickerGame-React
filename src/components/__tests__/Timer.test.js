
import React from 'react';
import ReactDom from 'react-dom';
import {render,fireEvent,cleanup,screen,container,act} from '@testing-library/react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import "@testing-library/jest-dom/extend-expect";
import {shallow,configure} from 'enzyme';
import renderer from "react-test-renderer";

it("Passes variables correctly from counter to Timer",()=>{
    const {getByTestId}=render(<Timer TimerSeconds="10"/>);
    expect(getByTestId('TimerTestId')).toHaveTextContent("10");
})

it("Matches Snapshot",()=>{
  const tree=renderer.create(<Timer TimerSeconds="10"/>).toJSON();
  expect(tree).toMatchSnapshot();
})

it("Check if timer works counter",()=>{
  jest.useFakeTimers();
  //In Total has 10 seconds advance to 2 seconds should display 8
  const {getByText}=render(<Timer/>);
  act(() => {
      jest.advanceTimersByTime(1000);
     
    })
    
   // expect(getByText('9')).toBeInTheDocument();
})

import Timer from './../Timer/Timer';