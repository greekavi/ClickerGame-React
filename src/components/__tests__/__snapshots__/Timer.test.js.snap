// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div>
  <b
    className="Timer"
    data-testid="TimerTestId"
  >
    Timer : 
    10
  </b>
</div>
`;
